File: LeetCode/C++/100-199/122.cpp
#include <iostream>
#include <vector>
using namespace std;
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxAns = 0;
        for(int i = 1 ; i < prices.size();++i){
            if(prices[i] > prices[i-1]){
                maxAns += prices[i] - prices[i-1];
            }
        }
        
        return maxAns;
    }
};

int main() {
	
	vector<int> prices{7,1,5,3,6,4};
	Solution sol;
	cout << sol.maxProfit(prices) << endl;
	return 0;
}

File: LeetCode/C#/100-199/122.cs
using System;

namespace LeetCode122
{
    public class Solution {
		public int MaxProfit(int[] prices) {
			int maxAns = 0;
			for(int i = 1 ; i < prices.Length;++i){
				if(prices[i] > prices[i-1]){
					maxAns += prices[i] - prices[i-1];
				}
			}
			
			return maxAns;
		}
	}
	
    public class Program
    {
        public static void Main(string[] args)
        {
            int[] prices = new int[]{7,1,5,3,6,4};
            Console.WriteLine(new Solution().MaxProfit(prices));
            Console.Read();
        }
    }
}
